Space awareness for Elastic Defend (#1943)
elastic/docs-content: solutions/security/get-started/spaces-security-faq.md

---
applies_to:
  stack: preview 9.1
  serverless:
    security: all
products:
  - id: security
  - id: cloud-serverless
---

> # Spaces and {{elastic-sec}} FAQ [security-spaces-faq]

Reviewer: I think the title of this page (and throughout the content below) should be specific for Elastic Defend, since these are only applicable to that component from within Elastic Security. They do not apply to how spaces work for other Security areas (ex. SIEM, Timeline, etc)

This page introduces {{elastic-sec}} space awareness and answers frequently asked questions about how {{elastic-defend}} integration policies, {{elastic-endpoint}} artifacts, and {{elastic-endpoint}} response actions function when utilizing {{kib}} spaces.

::::{admonition} Key points
* Artifacts such as trusted applications, event filters, and response action history are scoped by space to provide granular control over access.
* Role-Based Access Control (RBAC) defines who can manage global and space-specific resources. Users can view, edit, or manage artifacts based on their role privileges and the space context. 
* You need the global management privilege to manage global artifacts (those not associated with specific policies).
:::: 

::::{note}

> {{elastic-sec}}'s space awareness works in conjunction with {{fleet}}'s space awareness. space awareness is enabled by default in both applications, but for {{stack}} deployments that existed prior to version 9.1, {{fleet}} requires you to manually “opt-in” so that existing data can become space aware. To learn more, refer to [Fleet roles and privileges](/reference/fleet/fleet-roles-privileges.md).

Reviewer: Same comment here: Linking to fleet privileges pages does not seem to make sense here. We should link instead to the Fleet page that describes how they support Spaces instead (if that exists)

“Orphan” response actions are those associated only with deleted integration policies. These response actions are not visible in the response action history log because it can't be determined whether your current space has visibility of the policy associated with the response actions.

**How can I access orphan response actions?**

To make orphan response actions visible in a given space, you can make an API call with the space ID where you want them to appear. Below are several examples:

::::{important}
in order to use this API, you need {{kib}}'s built-in `superuser` role.
::::

:::{dropdown} View current orphan response actions space id:

API call:
`GET /internal/api/endpoint/action/_orphan_actions_space`

Response: 
```
{
  "data": {
    "spaceId": "admin"
  }
}
```
:::

:::{dropdown} Update orphan response action space id:
API call:
```
POST /internal/api/endpoint/action/_orphan_actions_space
{
  "spaceId": "admin"
}
```

Response:
```
{
  "data": {
    "spaceId": "admin"
  }
}
```
:::

To remove the space ID where orphan response actions appear, call the API with an empty string for `spaceId`.

> ## Endpoint Detection Rules [spaces-security-faq-endpoint-detection-rules]

Reviewer 2: ```suggestion
## Endpoint protection rules [spaces-security-faq-endpoint-protection-rules]
```

> By default, prebuilt detection rules use an index pattern that may be too broad for use in a particular space. In order to ensure that the space only shows the desired data in that space, you may need to customize the rule.

Reviewer 2: ```suggestion
By default, [endpoint protection rules](/solutions/security/manage-elastic-defend/endpoint-protection-rules.md) use an index pattern that may be too broad for use in a particular space. In order to ensure that the space only shows the desired data in that space, you may need to customize the rule.
```
I thought I added this suggestion in my initial review, but looks like I missed it.

> For example, the {{elastic-defend}} rule has an index pattern that picks up all data sent to `logs-endpoint.alerts-*`. This index pattern would pick up all events sent by Elastic Defend which may not be desirable. 

Reviewer 2: ```suggestion
For example, the Endpoint Security ({{elastic-defend}}) rule has an index pattern that picks up all data sent to `logs-endpoint.alerts-*`. This index pattern would pick up all events sent by {{elastic-defend}}, which may not be desirable. 
```
